src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,306 @@
+        })
+      }
+      if (this.state.initialMaxDateRange != newProps.max_date_range){
+        this.setState({
+          initialMaxDateRange: newProps.max_date_range,
+          maxDateRange: momentProps.maxDateRange
+        })
+      }
+    }
+
+    render() {
+        const {setProps, fireEvent} = this.props
+        return (
+          <DateRangePicker
+            startDate={ this.state.startDate }
+            startDatePlaceholderText={ this.props.start_date_placeholder_text }
+            endDate={ this.state.endDate }
+            endDatePlaceholderText={ this.props.end_date_placeholder_text }
+            onDatesChange={({ startDate, endDate }) => {
+              this.setState({ startDate, endDate });
+              if (startDate != null) {
+                const startDateStr = startDate.format('YYYY-MM-DD');
+                if (setProps) {
+                  setProps({
+                    startDate: startDateStr
+                  });
+                }
+                if (fireEvent) {
+                  fireEvent('change');
+                }
+              }
+              this.props.start_date = startDate;
+              if (endDate != null) {
+                const endDateStr = endDate.format('YYYY-MM-DD');
+                if (setProps) {
+                  setProps({
+                    endDate: endDateStr
+                  });
+                }
+                if (fireEvent) {
+                  fireEvent('change');
+                }
+              }
+            }}
+            focusedInput={ this.state.focusedInput }
+            onFocusChange={focusedInput => this.setState({ focusedInput })}
+            isOutsideRange={date =>
+              date < this.state.minDateRange || date >= this.state.maxDateRange
+            }
+            showClearDates={ this.props.clearable }
+            disabled={ this.props.disabled }
+            keepOpenOnDateSelect={ this.props.stay_open_on_select }
+            reopenPickerOnClearDates={ this.props.reopen_calendar_on_clear }
+            initialVisibleMonth={() => {
+              if (this.state.startDate != null) {
+                return this.state.startDate
+              } else {
+                return this.state.initialVisibleMonth
+              }
+            }}
+            numberOfMonths={ this.props.number_of_months_shown }
+            withPortal={ this.props.with_portal }
+            withFullScreenPortal={ this.props.with_full_screen_portal }
+            firstDayOfWeek={ this.props.first_day_of_week }
+            minimumNights={ this.props.minimum_nights }
+            enableOutsideDays={ this.props.show_outside_days }
+            monthFormat={ this.props.month_format }
+            displayFormat={ this.props.display_format }
+          />
+        );
+    }
+}
+
+DatePickerRange.propTypes = {
+    id: PropTypes.string,
+
+    /**
+     * Specifies the starting date for the component, best practice is to pass
+     * value via datetime object
```

Comment:
and the rest of the components. i.e. replace "best practice is to pass value via datetime object" with "Accepts `datetime.datetime` objects or strings in the format 'YYYY-MM-DD'"